core/math/transform_2d.cpp

```diff
@@ -80,13 +80,14 @@ real_t Transform2D::get_rotation() const {
 }
 
 void Transform2D::set_rotation(real_t p_rot) {
-
+	Vector2 scale = get_scale();
 	real_t cr = Math::cos(p_rot);
 	real_t sr = Math::sin(p_rot);
 	elements[0][0] = cr;
 	elements[0][1] = sr;
 	elements[1][0] = -sr;
 	elements[1][1] = cr;
+	set_scale(scale);
```

Comment:
What's the rationale for this change? It breaks compat, so it would be good to justify it a bit :)

Reply:
With the old code, the existing basis values were not used, such that it always ends up with a scale of 1. This change makes it so that it takes the previous scale and applies it after setting rotation. Transforms that already had a scale of 1 shouldn't be affected.

If there is a desire for a method that sets rotation and deleting the existing scale, I could add that (and possibly change the existing calls to that) but it probably isn't a good idea for a method simply called `set_rotation` to delete the scale.

Reply:
Thanks for the explanation. @reduz confirmed that this is a bug worth fixing for 3.1 :)